crates/yielder/src/provider.rs

```diff
@@ -0,0 +1,142 @@
+use std::sync::Arc;
+
+use alloy_primitives::Address;
+use async_trait::async_trait;
+use primitives::{AssetId, Chain};
+
+use crate::yo::YieldError;
+
+#[derive(Debug, Clone)]
+pub struct Yield {
+    pub name: String,
+    pub asset: AssetId,
+    pub provider: String,
+    pub apy: Option<f64>,
+}
+
+impl Yield {
+    pub fn new(name: impl Into<String>, asset: AssetId, provider: impl Into<String>, apy: Option<f64>) -> Self {
+        Self {
+            name: name.into(),
+            asset,
+            provider: provider.into(),
+            apy,
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct YieldTransaction {
+    pub chain: Chain,
+    pub from: String,
+    pub to: String,
+    pub data: String,
+    pub value: Option<String>,
+}
+
+#[derive(Debug, Clone)]
+pub struct YieldDetailsRequest {
+    pub asset: AssetId,
+    pub wallet_address: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct YieldDetails {
+    pub asset: AssetId,
+    pub provider: String,
+    pub share_token: String,
+    pub asset_token: String,
+    pub share_balance: Option<String>,
+    pub asset_balance: Option<String>,
+    pub apy: Option<f64>,
+    pub rewards: Option<String>,
+}
+
+impl YieldDetails {
+    pub fn new(asset: AssetId, provider: impl Into<String>, share_token: Address, asset_token: Address) -> Self {
+        Self {
+            asset,
+            provider: provider.into(),
+            share_token: share_token.to_string(),
+            asset_token: asset_token.to_string(),
+            share_balance: None,
+            asset_balance: None,
+            apy: None,
+            rewards: None,
+        }
+    }
+}
+
+#[async_trait]
+pub trait YieldProvider: Send + Sync {
+    fn protocol(&self) -> &'static str;
+    fn yields(&self, asset_id: &AssetId) -> Vec<Yield>;
+    async fn deposit(&self, asset: &AssetId, wallet_address: &str, amount: &str) -> Result<YieldTransaction, YieldError>;
```

Comment:
```suggestion
    async fn deposit(&self, asset: &AssetId, wallet_address: &str, value: &str) -> Result<YieldTransaction, YieldError>;
```

always `value` for big int values, amount for decimal values